### PR TITLE
Dotted and solid line mix with icon example

### PR DESCRIPTION
**Pull request by langsmith**
This pr is pretty simple. Shows two different lines connected to a `SymbolLayer` icon.

ezgif com-gif-maker

 http://turfjs.org/docs/#greatCircle isn't available for Java, so the lines are straight 😕 

This example was inspired by a seatback map I saw on a recent flight:

IMG_20190323_172126
IMG_20190323_172148

### CODE REVIEW

**Comment by tobrun** [comment]
> http://turfjs.org/docs/#greatCircle isn't available for Java, so the lines are straight confused

https://github.com/mapbox/mapbox-java/issues/1002

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/DottedAndSolidLineIconActivity.java [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+      public void onMapReady(@NonNull MapboxMap mapboxMap) {
+        mapboxMap.setStyle(Style.SATELLITE_STREETS, new Style.OnStyleLoaded() {
+          @Override
+          public void onStyleLoaded(@NonNull Style style) {
+
+            // Add the plane SymbolLayer icon to the map
+            style.addImage("plane-icon-id", BitmapFactory.decodeResource(
+              getResources(), R.drawable.ic_action_plane));
+            style.addSource(new GeoJsonSource("plane-source-id", MINNEAPOLIS_CITY_AIRPORT_POINT));
+            style.addLayer(new SymbolLayer("plane-symbol-layer-id", "plane-source-id").withProperties(
```
Comment (tobrun) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/DottedAndSolidLineIconActivity.java]: duplicate "plane-source-id", use constant instead

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/DottedAndSolidLineIconActivity.java [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+              iconImage("plane-icon-id"),
```
Comment (tobrun) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/DottedAndSolidLineIconActivity.java]: duplicate "plane-icon-id", use constant instead

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/DottedAndSolidLineIconActivity.java [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+            List<Point> solidLineList = new ArrayList<>();
+            solidLineList.add(MINNEAPOLIS_CITY_AIRPORT_POINT);
+            solidLineList.add(SAN_FRANCISCO_AIRPORT_POINT);
+            style.addSource(new GeoJsonSource("solid-line-source-id",
+              Feature.fromGeometry(LineString.fromLngLats(solidLineList))));
+            style.addLayerBelow(new LineLayer("solid-line-layer-id", "solid-line-source-id")
+              .withProperties(lineWidth(4.5f),
+                lineColor(Color.BLUE)), "plane-symbol-layer-id");
+          }
+        });
```
Comment (tobrun) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/DottedAndSolidLineIconActivity.java]: we are creating 3 source objects can we merge them to be one? For points vs lines nothing needs to be done but for solid/dot list we will need to add some kind of filter.

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/DottedAndSolidLineIconActivity.java (hunk 2 above)
Comment (langsmith) [reply MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/DottedAndSolidLineIconActivity.java]: fixed

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/DottedAndSolidLineIconActivity.java (hunk 1 above)
Comment (langsmith) [reply MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/DottedAndSolidLineIconActivity.java]: fixed

**Comment by langsmith** [comment]
Going to close. Don't think this is worthy of the demo app 😕 . Just a mix of LineLayer and SymbolLayer

**Pull request closed without merging**